src/tagmanager/tm_source_file.h

```diff
@@ -42,7 +42,11 @@ TMSourceFile *tm_source_file_new(const char *file_name, const char *name);
 
 void tm_source_file_free(TMSourceFile *source_file);
 
-gchar *tm_get_real_path(const gchar *file_name);
+gchar *tm_get_real_path(const gchar *file_name)
+#ifndef GEANY_PRIVATE
+G_DEPRECATED_FOR(utils_get_real_path)
+#endif
```

Comment:
We might need TM to depend on Geany anyway - while working on https://github.com/geany/geany/pull/963 I realized I'd need to know in TM if GIO is used for opening/saving files or not. 

We actually don't need to create the libtagmanager library - the sources inside "tagmanager" can just be other sources of libgeany.

Otherwise +1 for moving this function outside of TM - it doesn't belong there and I wanted to do this myself (but like Colomban I'd prefer having the implementation in utils).